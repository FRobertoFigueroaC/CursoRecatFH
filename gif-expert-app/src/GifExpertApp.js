import React, {useState} from 'react';
import {AddCategory} from "./components/AddCategory";

export const GiftExpertApp = () => {

    const [categories, setCategories] = useState(['Terror', 'Comedia', 'Drama']);


    return(
        <>
            <h2>GiftExpertApp</h2>
            {/*<AddCategory setCategories={setCategories}/>*/}
            <AddCategory/>
            <hr/>

            <ol>
                {
                    categories.map((category,i) => <li key={i}>{category}</li>)
                }
            </ol>
        </>
    )
}

export default GiftExpertApp;