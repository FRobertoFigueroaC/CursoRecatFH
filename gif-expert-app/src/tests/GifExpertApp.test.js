

import React from 'react';
import { shallow } from "enzyme";
import { GiftExpertApp } from "../GifExpertApp";

describe('Pruebas de GiftExpertApp ', () => {


    test('debe conicidir con snapshot', () => {
        const wrapper = shallow(<GiftExpertApp/>);     

        expect( wrapper).toMatchSnapshot();
    });

    test('debe mostrar una lista de categorias', () => {
        
        const categorias = ['Star Wars', 'Batman'];

        const wrapper = shallow(<GiftExpertApp defaultCategories={categorias}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);
    })
    
    
    
});
