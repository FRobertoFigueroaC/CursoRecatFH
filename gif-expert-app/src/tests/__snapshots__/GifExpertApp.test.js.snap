// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GiftExpertApp  debe conicidir con snapshot 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas de GiftExpertApp  debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Star Wars"
      key="Star Wars"
    />
    <GifGrid
      category="Batman"
      key="Batman"
    />
  </ol>
</Fragment>
`;
